checker/src/org/checkerframework/checker/lock/wholefiles/LockAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,261 @@
+
+    /** Annotation constants */
+    protected final AnnotationMirror LOCKHELD, LOCKPOSSIBLYHELD, SIDEEFFECTFREE, GUARDEDBY, JCIPGUARDEDBY, JAVAXGUARDEDBY, GUARDSATISFIED;
+
+    private final Class<? extends Annotation> checkerHoldingClass = Holding.class;
+
+    // Note that Javax and JCIP @GuardedBy is used on both methods and objects. For methods they are
+    // equivalent to the Checker Framework @Holding annotation.
+    private final Class<? extends Annotation> javaxGuardedByClass = javax.annotation.concurrent.GuardedBy.class;
+    private final Class<? extends Annotation> jcipGuardedByClass = net.jcip.annotations.GuardedBy.class;
+
+    private final Class<? extends Annotation> checkerLockingFreeClass = LockingFree.class;
+    private final Class<? extends Annotation> checkerReleasesNoLocksClass = ReleasesNoLocks.class;
+    private final Class<? extends Annotation> checkerMayReleaseLocksClass = MayReleaseLocks.class;
+    private final Class<? extends Annotation> sideEffectFreeClass = SideEffectFree.class;
+    private final Class<? extends Annotation> pureClass = Pure.class;
+
+    public LockAnnotatedTypeFactory(BaseTypeChecker checker, boolean useFlow) {
+        super(checker, useFlow);
+
+        LOCKHELD = AnnotationUtils.fromClass(elements, LockHeld.class);
+        LOCKPOSSIBLYHELD = AnnotationUtils.fromClass(elements, LockPossiblyHeld.class);
+        SIDEEFFECTFREE = AnnotationUtils.fromClass(elements, SideEffectFree.class);
+        GUARDEDBY = AnnotationUtils.fromClass(elements, GuardedBy.class);
+        JCIPGUARDEDBY = AnnotationUtils.fromClass(elements, net.jcip.annotations.GuardedBy.class);
+        JAVAXGUARDEDBY = AnnotationUtils.fromClass(elements, javax.annotation.concurrent.GuardedBy.class);
+        GUARDSATISFIED = AnnotationUtils.fromClass(elements, GuardSatisfied.class);
+
+        addAliasedAnnotation(javax.annotation.concurrent.GuardedBy.class, GUARDEDBY);
+        addAliasedAnnotation(net.jcip.annotations.GuardedBy.class, GUARDEDBY);
+
+        // This alias is only true for the Lock Checker. All other checkers must
+        // ignore the @LockingFree annotation.
+        addAliasedDeclAnnotation(LockingFree.class,
+                SideEffectFree.class,
+                AnnotationUtils.fromClass(elements, SideEffectFree.class));
+
+        // This alias is only true for the Lock Checker. All other checkers must
+        // ignore the @ReleasesNoLocks annotation.  Note that ReleasesNoLocks is
+        // not truly side-effect-free even as far as the Lock Checker is concerned,
+        // so there is additional handling of this annotation in the Lock Checker.
+        addAliasedDeclAnnotation(ReleasesNoLocks.class,
+                SideEffectFree.class,
+                AnnotationUtils.fromClass(elements, SideEffectFree.class));
+
+        postInit();
+    }
+
+    @Override
+    protected void postInit() {
+        super.postInit();
+
+        addInheritedAnnotation(AnnotationUtils.fromClass(elements,
```

Comment:
Is there a reason that you add these here rather than use the meta-annotation @InheritedAnnotation?  I think it is preferred to use @InheritedAnnotation. 

Otherwise, Is there some reason these need to be added after postInit?  Note that postInit generates types from stubfiles, so this annotations will be not inherited for method declarations there.  

(Seems like the JavaDoc on addInheritedAnnotation and postInit need to be improved.  I'll do that)